spring-integration-core/src/main/java/org/springframework/integration/handler/TriggerMessageHandler.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.handler;
+
+import org.springframework.messaging.Message;
+
+/**
+ * Classes implementing this interface can take some action when a trigger {@link Message}
+ * is received.
+ *
+ * @author Gary Russell
+ * @since 4.2
+ *
+ */
+public interface TriggerMessageHandler {
```

Comment:
How about `TriggerActionAware`, or `TriggerAction`?
It isn't `MessageHandler` anyway...